v3-sdk/trading/src/libs/trading.ts

```diff
@@ -151,30 +147,32 @@ async function getOutputQuote(route: Route<Currency, Currency>) {
   return ethers.utils.defaultAbiCoder.decode(['uint256'], quoteCallReturnData)
 }
 
-export async function getTokenTransferApprovals(
-  provider: ethers.providers.Provider,
-  tokenAddress: string,
-  fromAddress: string
+export async function getTokenTransferApproval(
+  token: Token
 ): Promise<TransactionState> {
-  if (!provider) {
+  const provider = getProvider()
+  const address = getWalletAddress()
+  if (!provider || !address) {
     console.log('No Provider Found')
     return TransactionState.Failed
   }
 
   try {
-    const tokenContract = new ethers.Contract(tokenAddress, ERC20_ABI, provider)
+    const tokenContract = new ethers.Contract(
+      token.address,
+      ERC20_ABI,
+      provider
+    )
 
     const transaction = await tokenContract.populateTransaction.approve(
       SWAP_ROUTER_ADDRESS,
-      JSBI.BigInt(AMOUNT_TO_APPROVE).toString()
+      TOKEN_AMOUNT_TO_APPROVE_FOR_TRANSFER
```

Comment:
This worked on the others, there's no special need on this contract for the JSBI to string step is there?